From 26d5e701657ab5b060f986a73ba10214beb83abf Mon Sep 17 00:00:00 2001
From: Bamvor Jian Zhang <bjzhang@suse.com>
Date: Tue, 10 Dec 2013 16:22:48 +0800
Subject: [PATCH 4/5] improve ao: move libxl_event_register_callbacks before
 libxl_domain_create_xxx in order to support ao in create


Signed-off-by: Bamvor Jian Zhang <bjzhang@suse.com>
---
 src/libxl/libxl_driver.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libxl/libxl_driver.c b/src/libxl/libxl_driver.c
index a44800b..ebf93da 100644
--- a/src/libxl/libxl_driver.c
+++ b/src/libxl/libxl_driver.c
@@ -524,8 +524,6 @@ libxlCreateDomEvents(virDomainObjPtr vm)
 {
     libxlDomainObjPrivatePtr priv = vm->privateData;
 
-    libxl_event_register_callbacks(priv->ctx, &ev_hooks, vm);
-
     if (libxl_evenable_domain_death(priv->ctx, vm->def->id, 0, &priv->deathW))
         goto error;
 
@@ -713,6 +711,8 @@ libxlVmStart(libxlDriverPrivatePtr driver, virDomainObjPtr vm,
         goto error;
     }
 
+    libxl_event_register_callbacks(priv->ctx, &ev_hooks, vm);
+
     if (ao_how_enable && async) {
         ao_how_init(vm);
     }
-- 
1.8.1.4

